### PR TITLE
Cleanup macros and CfgPatches, Merge game component into main

### PR DESCRIPTION
**Pull request by jonpas**
**When merged this pull request will:**
- Removed unused and duplicate macros.
- Remove macros already defined in CBA.
- Merge `game` component into `main` (`lib_` functions need to be separated into files and put into `sys_core` component - marked with `@todo`).
- Add debug defines to all components (inactive/commented).
- Change all `QUOTE(GVAR())` to `QGVAR()`
- Change `ARR_n` macros to CBA variant.
- Add beautified component names (displayed in Eden).
- Streamline `CfgPatches` with URL, author(s) and full version config.
- Remove a lot of whitespace between macros and their values (switch to spaces broke most of their alignment anyways and any proper editor has nice highlighting to differentiate).
- Use CBA log macros.
- Add `ACRE_DEPRECATED` macro and mark deprecated function with that macro and removal in 2.5.0.

Beautified component name in action:

CBA log macros and CBA versioning (from #56) in action:
54d771ce-876f-11e6-8127-d34109fbd32b

### CODE REVIEW

File: addons/api/fnc_mapChannelFieldName.sqf [hunk 1]
```diff
@@ -16,7 +16,9 @@
  */
 #include "script_component.hpp"
 
-params["_radioId", "_fieldName"];
+ACRE_DEPRECATED(QFUNC(mapChannelFieldName),"2.5.0","None");
```
Comment (jonpas) [inline addons/api/fnc_mapChannelFieldName.sqf]: It's actually used in 2 places, not sure what to do... doesn't look deprecated to me?
- https://github.com/IDI-Systems/acre2/blob/f2bdfe166e162ee23e7aad143fa9d72f56e6c28c/addons/api/fnc_getPresetChannelField.sqf#L52
- https://github.com/IDI-Systems/acre2/blob/f2bdfe166e162ee23e7aad143fa9d72f56e6c28c/addons/api/fnc_setPresetChannelField.sqf#L67

Comment (Soldia1138) [reply addons/api/fnc_mapChannelFieldName.sqf]: "compat" of course.

Comment (jonpas) [reply addons/api/fnc_mapChannelFieldName.sqf]: Ah I see, will fix the header then, thanks! 👍

**Review by Sniperhid — APPROVED** [review]
LGTM

**Pull request merged**